City-of-Helsinki/berth-reservations-admin
# VEN-860, VEN-876, VEN-881 | Improve invoice sending UX

## Description :sparkles:

- Disable send invoice button once an invoice has been sent
- Add success toast when invoice sending has been processed
- Remove "send invoice" button once invoice has been further processed
- Display "invoice sent" as button label when offer is still pending user interaction
- UnmarkedWsNoticeView: disable buttons that are waiting for server response
- 

### Closes :no_good_woman:

[VEN-860](https://helsinkisolutionoffice.atlassian.net/browse/VEN-860)
[VEN-876](https://helsinkisolutionoffice.atlassian.net/browse/VEN-876)
[VEN-881](https://helsinkisolutionoffice.atlassian.net/browse/VEN-881)

## Testing :alembic:

### Automated tests :gear:️

- `src/features/invoiceCard/__tests__/InvoiceCard.test.tsx`
- ` src/common/deleteButton/__tests__/DeleteButton.test.tsx`
- Updated relevant tests.

### Manual testing :construction_worker_man:

- Navigate to "Nostojärjestysilmoitukset"
- Depending on application status, the send invoice button should be displayed as:
  - "Käsitelty": button should be visible and enabled, with text "Lähetä lasku"
  - "Lähetetty": button should be disabled, with text "Tarjous lähetetty"
  - Any other status: should not display the button at all.
- Depending on application status, the delete invoice button should be displayed as:
  - "Käsitelty": button should be visible and enabled, pressing the button should open a confirmation dialog
  - Any other status: should not display the button at all.
- Sending an invoice should display a success toast.
- Deletion button should not be present for WS notices with application status other than "Käsitelty"/"Odottaa käsittelyä"
